crypto/params.c

```diff
@@ -13,6 +13,16 @@
 #include "internal/thread_once.h"
 #include "internal/numbers.h"
 
+/*
+ * Return the number of bits in the mantissa of a double.  This is used to
+ * shift a larger integral value to determine if it will exactly fit into a
+ * double.
+ */
+static unsigned int real_shift(void)
+{
+    return sizeof(double) == 4 ? 24 : 53;
```

Comment:
It's both.  The mantissa field in an IEEE double is 52 bits in size.  The 53rd bit is implied since it will always be a _1_ after normalisation.

With the uptake of the IEEE standard, double should always be 4 or 8 bytes.  I've not seen a platform where IEEE wasn't used for decades (Vax has it's own floating point formats but double was still 8 bytes worked).

A check in `sanitytest.c` might be sensible, not for this PR though.  There are assumptions about four and eight byte integers and these aren't checked.

Reply:
This change shouldn't be here.